it/src/main/java/com/google/cloud/teleport/it/common/ResourceManagerUtils.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.common;
+
+import static com.google.common.hash.Hashing.goodFastHash;
+import static java.lang.Math.min;
+import static org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.hash.HashFunction;
+import com.google.re2j.Pattern;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Random;
+
+/** Common utilities for ResourceManager implementations. */
+public class ResourceManagerUtils {
```

Comment:
just fyi this will clash with https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/448/, but merging should be quite simple